lib/kafo/param.rb

```diff
@@ -120,12 +120,8 @@ def interpret_validation_args(args)
       args.map do |arg|
         if arg.to_s == "$#{self.name}"
           self.value
-        elsif arg.class.name == 'Puppet::Parser::AST::ASTArray'
-          interpret_validation_args(arg.to_a)
-        elsif arg.class.name == 'Puppet::Parser::AST::Concat'
-          interpret_validation_args(arg.value).join
-        elsif arg.respond_to? :value
-          arg.value
+        elsif arg.is_a? String
+          arg.gsub("$#{self.name}", self.value.to_s)
```

Comment:
An example validation that would trigger both parts is:

    validate_re($db_type, ["^mysql$", "^sqlite$"], "invalid $db_type DB type")

The first argument is passed back from kafo_parsers as `"$db_type"` and would be replaced by the first condition.

The third argument is passed back from kafo_parsers as `"invalid $db_type DB type"`, so is matched and replaced with gsub by this condition.

I could probably remove the first condition entirely now.